tcs-persistence/src/test/java/com/transformuk/hee/tis/tcs/service/repository/PostRepositoryTest.java

```diff
@@ -172,6 +173,72 @@ public void findPostsAndPlacementsByProgrammeIdAndSpecialtyIdShouldFindPostsLink
               .getForenames()));
 
     }
+  }
+
+  @Test
+  public void testClearOldPostReferencesInPostTableWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setOldPost(post1);
+    entityManager.persist(post2);
+    entityManager.flush();
+
+    postRepository.clearOldPostReferences(post1.getId());
+    entityManager.flush();
+    entityManager.clear();
+
+    Post updatedPost = entityManager.find(Post.class, post2.getId());
+    Assert.assertNull(updatedPost.getOldPost());
+  }
+
+  @Test
+  public void testClearNewPostReferencesInPostTableWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setOldPost(post1);
+    entityManager.persist(post2);
+    entityManager.flush();
+
+    postRepository.clearNewPostReferences(post1.getId());
+    entityManager.flush();
+    entityManager.clear();
+
+    Post updatedPost = entityManager.find(Post.class, post2.getId());
+    assertNull(updatedPost.getNewPost());
+  }
+
+  @Test
+  public void testClearPostReferencesForBothOldAndNewPostWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setOldPost(post1);
+    post2.setNewPost(post1);
```

Comment:
This condition doesn't make logical sense.
If you have `post2.setOldPost(post1);` then next condition should be `post1.setNewPost(post2);`.

Reply:
Changed